tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/HttpClient/HttpMessageHandlerCommon.cs

```diff
@@ -19,20 +19,23 @@ internal static class HttpMessageHandlerCommon
         public static CallTargetState OnMethodBegin<TTarget, TRequest>(TTarget instance, TRequest requestMessage, CancellationToken cancellationToken, IntegrationId integrationId, IntegrationId? implementationIntegrationId)
             where TRequest : IHttpRequestMessage
         {
+            if (requestMessage.Instance is null)
```

Comment:
Just wondering, given that we expect this `if` to basically always be false, doesn't this mean we basically _always_ introduce a branch, compared to the way it was originally written? 

I prefer the "guard clause" approach too, just wondered, given this is an "improve performance PR" 🙂

Reply:
I've introduced this changed because we were boxing `requestMessage.Headers` twice (internally is an struct that is returned as an interface). Branching is not avoidable, the previous one might be a little bit better due to the branch predictor but it will depend of the generated code from the jit.